### ISSUE TITLE
--enable-gzip break gke ingress / argocd ui

### ISSUE THREAD

**Issue by bradenwright-opunai** [argoproj/argo-cd#14973]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**
setting the argument for `--enable-gzip` break the UI so I can'r see any Applications, AppProjects or Repositories.  If I port forward to the argocd-server it works fine (and is faster which is my goal).  Not sure what about the GCP LB is breaking the UI / Ingress.

**To Reproduce**

Deploy ArgoCD to GKE using directions for GKE ingress https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#google-cloud-load-balancers-with-kubernetes-ingress ensure that ingress / argocd work, then add the argument for --enabled-gzip, if using helm set:

```
server:
  extraArgs:
    - --insecure
    - --enable-gzip
``` 
--insecure is optional but I'm using SSL at the LB

**Expected behavior**

That I can see applications, repos, appprojects in the ArgoCD UI when going thru ingress / gcp lb, not just when port-forwarding

**Screenshots**

Not 100% sure how to add screenshots but I do have some

**Version**

```shell
v2.7.6+00c914a.dirty
```

**Comment by bradenwright-opunai** [comment]
Any update on this

**Comment by crenshaw-dev** [comment]
I think we need more info.

Are you getting console errors in your web browser? Error responses on network requests?

Is the Argo CD API server logging any errors? Are we confident that the requests are even reaching Argo CD?

**Comment by bradenwright-opunai** [comment]
Here's some screenshots of errors, more than happy to provide anything else:

<img width="1721" alt="Screenshot 2023-08-09 at 12 11 16 AM" src="https://github.com/argoproj/argo-cd/assets/96085456/7665daaf-d63a-4d52-a5d7-ae55f9f6b8e1">
<img width="454" alt="Screenshot 2023-08-09 at 12 51 23 AM" src="https://github.com/argoproj/argo-cd/assets/96085456/6c69320a-b8ff-458d-a007-2cfaa4e8335a">
<img width="587" alt="Screenshot 2023-08-09 at 12 52 31 AM" src="https://github.com/argoproj/argo-cd/assets/96085456/760540a9-502b-45ca-845f-8500a056ae9e">

Here's a screenshot of it working (this was after removing gzip, I can get screenshots of it working via port-forward too if need):
<img width="1723" alt="Screenshot 2023-08-09 at 12 53 07 AM" src="https://github.com/argoproj/argo-cd/assets/96085456/5d7fdd9d-b057-49f6-a495-0db981ce3182">
<img width="864" alt="Screenshot 2023-08-09 at 12 53 19 AM" src="https://github.com/argoproj/argo-cd/assets/96085456/6304d0b0-d1b4-42ab-886d-ad4634470f45">

Here's some docs I found on GCP about troubleshooting:
https://cloud.google.com/load-balancing/docs/https/troubleshooting-ext-https-lbs#compression-not-working

**Comment by crenshaw-dev** [comment]
@bradenwright-opunai deleted the screenshots because they contained an Argo CD token for a publicly-accessible endpoint.

But I see you're getting 502. So looks like something between Argo CD and the client is rejecting the request.

**Comment by bradenwright-opunai** [comment]
Thanks, for deleting them and the reply.  Yeah i just don't really know what to check b/c it works fine if a port forward and it doesn't seem like there are any options on GCP side, the docs of it not working reference setting in the nginx.conf and things of that nature.  The only change I'm making is adding `--enabled-gzip` flag.

Any ideas on anytihing to try to prove things out more, or what maybe going wrong.

**Comment by bradenwright-opunai** [comment]
So this gets a little more interesting... I tried to deploy similar code in another GCP project to reproduce the issue and test somewhere that wasn't in use.  The new install didn't have an issue, I thought maybe it was bc I used the latest version of Argocd:

`v2.7.6+00c914a.dirty` to `v2.8.0+804d4b8`

and it bumps the chart version to:
`5.36.7` to `5.43.4`

I then upgraded my existing install to versions: `v2.8.0+804d4b8` and `5.43.4` but still I'm getting the same error / issue.  I'm gonna compare my 2 installs and try to reproduce the issue.   But it's feeling more like an argocd issue than a GCP LB right now.  I also tried to recreate the GCP LB / K8s ingress just to be safe, but it didn't help.
